master/buildbot/test/unit/test_schedulers_base.py

```diff
@@ -158,9 +158,10 @@ def gotChange(got_change, got_important):
         d = sched.startConsumingChanges(**kwargs)
 
         def test(_):
-            # check that it registered a callback
-            self.assertEqual(len(self.mq.qrefs), 1)
-            qref = self.mq.qrefs[0]
+            # check that it registered callbacks
+            self.assertEqual(len(self.mq.qrefs), 2)
+
+            qref = self.mq.qrefs[1]
             self.assertEqual(qref.filter, ('changes', None, 'new'))
```

Comment:
You can call the callback directly from the tests instead of using the mq api to send the message. you need to make your test inlineCallbacks and then wait for the results of the callback in the test and make the appropriate assertion after that.
There are plenty of examples in the current tests